### PR TITLE
Feature/link network request data

### PR DESCRIPTION
**Pull request by DrewBradley**
### Is this a fix or a feature?
I'm really not sure. I want to say fix, but at this point I think I'm making things worse.

### What is the change/fix?
Make displayDirections function properly when using API data, since y'know they don't include the same data as the data we built the site with.

### Where should the reviewer start?
all of user.js. and lines 147 - 203 in ```scripts.js```

### How should this be tested?
Open scripts.js. Look at it.

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -2,24 +2,24 @@
 import './css/base.scss';
 import './css/styles.scss';
 
-import recipeData from './data/recipes';
-import ingredientsData from "./data/ingredients"
-
```
Comment (elsafluss) [inline src/scripts.js:7]: Feels good to get rid of these, means we might have learned something.

File: src/scripts.js [hunk 2]
```diff
@@ -42,56 +45,55 @@ function onStartup() {
...
+
   const userName1 = document.querySelector(".user-name-small")
   userName1.innerHTML =
     user.name.split(" ")[0] + " " + user.name.split(" ")[1][0]
 }
 
-function favoriteCard(event) {
-  let recipe = cookbook.recipes.find((recipe) => {
-    if (recipe.id === Number(event.target.id)) {
+function favoriteCard(event, cookbook) {
```
Comment (elsafluss) [inline src/scripts.js:84]: We figured out how to / where to call the APIs. Go team.

File: src/scripts.js [hunk 3]
```diff
@@ -132,79 +121,87 @@ function updateStar(event) {
...
   } else if (event.target.classList.contains("card-picture")) {
     displayDirections(event)
-  } else if (event.target.classList.contains("add-button")) {
-    let recipe = cookbook.recipes.find((recipe) => {
-      if (recipe.id === Number(event.target.id)) {
-        return recipe
-      }
-    })
-    user.addToRecipeArray(recipe, user.recipesToCook)
+    // } else if (event.target.classList.contains("add-button")) {
```
Comment (elsafluss) [inline src/scripts.js:127]: Commented out code

File: src/scripts.js [hunk 4]
```diff
@@ -132,79 +121,87 @@ function updateStar(event) {
...
-    ${ingredient.quantity.amount.toFixed(2)} ${ingredient.quantity.unit}
-    ${ingredient.name}</li></ul>
-    `)
-  })
-  recipeObject.instructions.forEach(instruction => {
-    instructionsSpan.insertAdjacentHTML('beforebegin', `<li>
-    ${instruction.instruction}</li>
-    `)
-  })
+  cardArea.classList.add("all")
```
Comment (elsafluss) [inline src/scripts.js:148]: This one was so tricksy. WOOOO

File: src/user.js [hunk 5]
```diff
@@ -7,7 +7,7 @@ class User {
     this.favoriteRecipes = [];
     this.recipesToCook = [];
     this.shoppingList = [];
-    this.ingredientsData = ingredientsData;
+    this.ingredientsData = ingredientsData || [];
```
Comment (colefiscus) [inline src/user.js:10]: Can this just be [ ] ?

File: src/util.js [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
-import ingredientsData from "./data/ingredients"
+import Cookbook from './cookbook';
 
 function createRandomUser(users) {
   let userId = (Math.ceil(Math.random() * 49) + 1)
   let newUser = users.find(user => {
     return user.id === Number(userId);
   });
-  let user = new User(userId, newUser.name, newUser.pantry, ingredientsData)
+  let user = new User(userId, newUser.name, newUser.pantry)
```
Comment (elsafluss) [inline src/util.js:9]: sneaky

**Review by colefiscus — APPROVED** [review]
Just some commented out code to cut out.

File: src/scripts.js (hunk 3 above)
Comment (DrewBradley) [reply src/scripts.js:127]: I was trying to handle all the displayDirections.

File: src/user.js (hunk 5 above)
Comment (DrewBradley) [reply src/user.js:10]: fax.

Comment (elsafluss) [reply src/user.js:10]: maybe?

File: src/scripts.js (hunk 3 above)
Comment (DrewBradley) [reply src/scripts.js:127]: We are using this code later if we get to displaying the added recipes to cook in the array attached to the user.

**Pull request merged**